### PR TITLE
Security and zkApps

### PR DESCRIPTION
**Pull request by mitschabaude**
This adds a new section under "Creating zkApps", called "Security and zkApps".

Topics covered:
* Advice for auditing zkApps
* Attack model to understand
* Example of introducing vulnerabilities in a zkApp
* List of best security practices

I didn't get to write a final section on considerations in low-level circuit writing. This is its whole own can of worms and will need some work to present well.

In addition, this adds a small section on the o1js intro page to host our internal audit and future audits of o1js.

### CODE REVIEW

File: docs/zkapps/secure-zkapps.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  - o1js
+  - mina protocol
+  - zero knowledge proofs
+---
+
+:::info
+
+zkApp programmability is not yet available on the Mina Mainnet, but zkApps can now be deployed on Berkeley Testnet.
+
+:::
```
Comment (shimkiv) [inline docs/zkapps/secure-zkapps.mdx:18]:
```suggestion
:::info

zkApp programmability is not yet available on the Mina **Mainnet**, but zkApps can now be deployed on the Mina **Devnet**.

:::
```

Comment (mitschabaude) [reply docs/zkapps/secure-zkapps.mdx:18]: shouldn't we just change this for all pages at once?

File: docs/zkapps/o1js/index.mdx [hunk 2]
```diff
@@ -43,7 +43,7 @@ function knowsPreimage(preimage: Field) {
 }
 
 const expectedHash =
-  Field(0x1d444102d9e8da6d566467defcc446e8c1c3a3616d059facadbfd674afbc37ecn);
+  0x1d444102d9e8da6d566467defcc446e8c1c3a3616d059facadbfd674afbc37ecn;
```
Comment (ymekuria) [inline docs/zkapps/o1js/index.mdx:46]:
```suggestion
  '0x1d444102d9e8da6d566467defcc446e8c1c3a3616d059facadbfd674afbc37ecn';
```

Comment (ymekuria) [reply docs/zkapps/o1js/index.mdx:46]: Should this be a string?

File: docs/zkapps/secure-zkapps.mdx [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+The pattern of passing in the full `AccountUpdate` here, and not just amount and address, is good practice and more flexible than creating the account updates inline:
+It allows the method to be used by zkApps, not just typical end-user accounts.
+zkApps need to put their own proof on the account update to authorize a spend.
+
+:::
+
+### Creating an insecure contract
+
+We need to use either `assertCanMint()` or `assertCanBurn()`, but how do we know which one? Well, let's just add a parameter to the method that tells us whether this is a mint or a burn. Then let's call the appropriate method based on that parameter. Github Copilot fills this out nicely for us:
```
Comment (ymekuria) [inline docs/zkapps/secure-zkapps.mdx:103]: "Github Copilot fills this out nicely" - LOL

**Review by ymekuria — APPROVED** [review]

**Comment by stevenpack** [comment]
Otherwise, LGTM

File: docs/zkapps/o1js/index.mdx (hunk 2 above)
Comment (mitschabaude) [reply docs/zkapps/o1js/index.mdx:46]: no, bigint literal!

**Pull request merged**